### ISSUE TITLE
Captured screenshots on failure being coerced to `runner`

### ISSUE THREAD

**Issue by svobom57** [cypress-io/cypress#5920]
### Current behavior:
When my test fails, the screenshot taken is always captured in the `runner` size as per the documentation: https://docs.cypress.io/api/cypress-api/screenshot-api.html#Arguments:
>  For screenshots automatically taken on test failure, capture is always coerced to runner.

 I was curious, why that's the case? I think it'd be beneficial to be able to override this configuration even for the screenshot on failure:
```js
Cypress.Screenshot.defaults({
  capture: 'fullPage'
})
```
Often times I find myself looking at a failure video where Cypress can't find an element and I'm only seeing a portion of the page which isn't really helpful

### Desired behavior:
When I configure Cypress to capture screenshots like so:
```js
Cypress.Screenshot.defaults({
  capture: 'fullPage'
})
```

I expect Cypress to take screenshot of the Full Page on failure rather than always taking `runner` despite the configuration.

### Versions
Cypress `3.7.0`, MacOS: `10.14.6`

**Comment by karthi07** [comment]
Can I work on this ?

**Comment by jennifer-shehane** [comment]
@karthi07 Yes we would love an open PR - even one that is a work in progress is fine. Check out our [contributing doc](https://github.com/cypress-io/cypress/blob/develop/CONTRIBUTING.md) and ask us if you get stuck.

**Comment by Seitar18** [comment]
Any news?
Still actual.

**Comment by kizylirio** [comment]
Any idea about how bypass it?

**Comment by johnnylopes** [comment]
any workaround for this? @karthi07

**Comment by astone123** [comment]
So I think there are two different ways that we could go about offering a configuration for this:

1. Add a new configuration option to `screenshot`, possibly `captureOnRunFailure` which takes the same options as `capture`

or

2. Update the `screenshotOnRunFailure` option to accept values for `screenshot.capture` (`runner`, `fullPage`, `viewport`, or `off`), rather than `true`/`false` (this would be a breaking change)

Routing this to e2e core

**Comment by thomasKuntermannDiva-e** [comment]
does the following also belongs to this thread or may I open another one(?):
- when using cy.screenshot({capture: 'viewport'}) and we have Cypress env settings like viewportWidth & viewportHeight, I would expect that screenshots are taken with the custom dimensions. But Cypress currently makes screenshots in the Cypress Runner dimensions, in our case 1280x720, and not our custom settings. Only "fullpage" fits to our requirements, to have a high resolution screenshot.
